packages/cubejs-pinot-driver/src/PinotQuery.ts

```diff
@@ -0,0 +1,165 @@
+  }
+}
+
+class PinotFilter extends BaseFilter {
+  public likeIgnoreCase(column: any, not: any, param: any, type: string) {
+    const p = (!type || type === 'contains' || type === 'ends') ? '%' : '';
+    const s = (!type || type === 'contains' || type === 'starts') ? '%' : '';
+    return `LOWER(${column})${not ? ' NOT' : ''} LIKE CONCAT('${p}', LOWER(${this.allocateParam(param)}) , '${s}') ESCAPE '\\'`;
+  }
+
+  public castParameter() {
+    if (this.definition().type === 'boolean') {
+      return 'CAST(? AS BOOLEAN)';
+    } else if (this.measure || this.definition().type === 'number') {
+      // TODO here can be measure type of string actually
+      return 'CAST(? AS DOUBLE)';
+    }
+
+    return '?';
+  }
+}
+
+export class PinotQuery extends BaseQuery {
+  public newFilter(filter: any): PinotFilter {
+    return new PinotFilter(this, filter);
+  }
+
+  public timeStampParam() {
+    return '?';
+  }
+
+  public timeStampCast(value: string) {
+    return `CAST(${value} as TIMESTAMP)`;
+  }
+
+  public dateTimeCast(value: string) {
+    return value;
+  }
+
+  public convertTz(field: string) {
+    return this.timeStampCast(`toDateTime(${field}, ${DATE_TIME_FORMAT}, '${this.timezone}')`);
+  }
+
+  public timeGroupedColumn(granularity: GRANULARITY_ID, dimension: string) {
+    return this.timeStampCast(`dateTrunc('${GRANULARITY_TO_INTERVAL[granularity]}', ${dimension})`);
+  }
+
+  public subtractInterval(date: string, interval: string) {
+    const [intervalValue, intervalUnit] = interval.split(' ');
+    return `${this.timeStampCast(date)} - fromEpoch${intervalUnit}(${intervalValue})`;
+  }
+
+  public addInterval(date: string, interval: string) {
+    const [intervalValue, intervalUnit] = interval.split(' ');
+    return `${this.timeStampCast(date)} + fromEpoch${intervalUnit}(${intervalValue})`;
+  }
+
+  public seriesSql(timeDimension: BaseTimeDimension) {
+    const values = timeDimension.timeSeries().map(
+      ([from, to]) => `select '${from}' f, '${to}' t`
+    ).join(' UNION ALL ');
+    return `SELECT ${this.timeStampCast('dates.f')} date_from, ${this.timeStampCast('dates.t')} date_to FROM (${values}) AS dates`;
+  }
+
+  public applyMeasureFilters(evaluateSql: '*' | string, symbol: any, cubeName: string) {
+    if (!symbol.filters || !symbol.filters.length) {
+      return evaluateSql;
+    }
+
+    const where = this.evaluateMeasureFilters(symbol, cubeName);
+
+    return `${evaluateSql === '*' ? '1' : evaluateSql}) FILTER (WHERE ${where}`;
+  }
+
+  /**
+   * @return {string}
+   */
+  public timestampFormat() {
+    return 'YYYY-MM-DD HH:mm:ss.SSS';
+  }
+
+  public unixTimestampSql() {
+    return this.nowTimestampSql();
+  }
+
+  public defaultRefreshKeyRenewalThreshold() {
+    return 120;
+  }
+
+  public defaultEveryRefreshKey() {
+    return {
+      every: '2 minutes'
+    };
+  }
+
+  public hllInit(sql: string) {
+    return this.countDistinctApprox(sql); // todo: ensure the correct way to do so in pinot
```

Comment:
This doesn't seem correct so let's remove it for now to avoid weird error messages